x-pack/plugins/security/server/lib/authorization/app_authorization.test.ts

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Server } from 'hapi';
+import { AuthorizationService } from './service';
+
+import { Feature } from 'x-pack/plugins/xpack_main/types';
+import { XPackMainPlugin } from 'x-pack/plugins/xpack_main/xpack_main';
+import { actionsFactory } from './actions';
+import { initAppAuthorization } from './app_authorization';
+
+const actions = actionsFactory({
+  get(key: string) {
+    if (key === 'pkg.version') {
+      return `1.0.0-zeta1`;
+    }
+
+    throw new Error(`Unexpected config key: ${key}`);
+  },
+});
+
+const createMockXPackMainPlugin = (): XPackMainPlugin => {
+  const features: Feature[] = [
+    {
+      id: 'foo',
+      name: 'Foo',
+      app: ['foo'],
+      privileges: {},
+    },
+  ];
+  return {
+    getFeatures: () => features,
+  } as XPackMainPlugin;
+};
+
+describe('initAppAuthorization', () => {
+  test(`route that doesn't start with "/app/" continues`, async () => {
+    const server = new Server();
+    initAppAuthorization(server, createMockXPackMainPlugin(), {} as AuthorizationService);
+    server.route([
+      {
+        method: 'GET',
+        path: '/api/foo',
+        handler: () => {
+          return 'foo app response';
+        },
+      },
+    ]);
+    const { result, statusCode } = await server.inject({
+      method: 'GET',
+      url: '/api/foo',
+    });
+    expect(result).toBe('foo app response');
+    expect(statusCode).toBe(200);
+  });
+
+  test(`protected route that starts with "/app/", but "mode.useRbacForRequest()" returns false continues`, async () => {
+    const server = new Server();
+    const mockAuthorizationService: AuthorizationService = {
+      mode: {
+        useRbacForRequest: jest.fn().mockReturnValue(false),
+      },
+    } as any;
+    initAppAuthorization(server, createMockXPackMainPlugin(), mockAuthorizationService);
+    server.route([
+      {
+        method: 'GET',
+        path: '/app/foo',
+        handler: () => {
+          return 'foo app response';
+        },
+      },
+    ]);
+    const { request, result, statusCode } = await server.inject({
+      method: 'GET',
+      url: '/app/foo',
+    });
+    expect(result).toBe('foo app response');
+    expect(statusCode).toBe(200);
+    expect(mockAuthorizationService.mode.useRbacForRequest).toHaveBeenCalledWith(request);
+  });
+
+  test(`unprotected route that starts with "/app/", and "mode.useRbacForRequest()" returns true continues`, async () => {
+    const headers = {
```

Comment:
Is the auth header useful for this test? It's not asserted anywhere, and omitting it doesn't fail, in my testing.

Reply:
Probably not, I will remove it.